lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -36,9 +36,16 @@ def visit(node)
       when YARP::ModuleNode
         add_index_entry(node, Index::Entry::Module)
       when YARP::ConstantWriteNode, YARP::ConstantOrWriteNode
-        add_constant(node)
-      when YARP::ConstantPathWriteNode, YARP::ConstantPathOrWriteNode
-        add_constant_with_path(node)
+        name = fully_qualify_name(node.name.to_s)
+        add_constant(node, name)
+      when YARP::ConstantPathWriteNode, YARP::ConstantPathOrWriteNode, YARP::ConstantPathOperatorWriteNode,
+        YARP::ConstantPathAndWriteNode
+
+        # ignore variable constants like `var::FOO` or `self.class::FOO`
+        return unless node.target.parent.nil? || node.target.parent.is_a?(YARP::ConstantReadNode)
+
+        name = fully_qualify_name(node.target.location.slice)
+        add_constant(node, name)
```

Comment:
Are we missing the handling of constant target nodes? `A, B = 42` and `FOO::A, FOO::B = 42`?